File: pyround1/788.旋转数字.py
#
# @lc app=leetcode.cn id=788 lang=python3
#
# [788] 旋转数字
#

# @lc code=start
class Solution:
    def rotatedDigits(self, N: int) -> int:
        ans = 0
        for x in range(1, N+1):
            S = str(x)
            ans += (all(d not in '347' for d in S) and any(d in '2569' for d in S))
        return ans
# @lc code=end

File: 788.旋转数字.cpp
/*
 * @lc app=leetcode.cn id=788 lang=cpp
 *
 * [788] 旋转数字
 */

// @lc code=start
class Solution {
public:
    int rotatedDigits(int N) {
        int count = 0;
        for(int i = 1; i <= N; i++){
            if(helper(i))
                count++;
        }
        return count;
    }
    bool helper(int val){
        vector<int> tmp;
        while(val){
            if(val <= 10)
            {
                tmp.push_back(val);
                break;
            }
            else{
                tmp.push_back(val%10);
                val /= 10;
            }
        }
        for(int i = 0; i < tmp.size(); i++){
            if(tmp[i] != 2 and tmp[i] != 5 and tmp[i] != 6 and tmp[i] != 9)
                return false;
        }
        tmp.clear();
        return true;
    }

};
// @lc code=end

